### DISCUSSION TITLE
Remove uploaded file

### DISCUSSION THREAD

**Discussion by sinnbeck** [filamentphp/filament#9180] [Help]
When a user uploads a file that fails validation I would live to remove it.
I have found the method (removeFormUploadedFile()) to remove it in the form code, but sadly it does not update the frontend.
```php
->afterStateUpdated(function ($state) {
                        try {
                            return $this->validateOnly('data.files');

                        } catch (ValidationException $exception) {
                            foreach ($state as $key => $file) {
                                $this->removeFormUploadedFile('data.files', $key);
                            }

                            throw $exception;
                        }
                    })
```
I know this code also removes previously uploaded files, but one problem at a time.

Has anyone found a way to remove a file and get the frontend to update to reflect this as well?

Another idea would be to visually mark the file that is "bad". If a user drags in 10 files, it might be hard for them to figure out which file was the failed one.

**Comment by nguyenhai97** [comment]
I have same problems
